tests/test_search.py

```diff
@@ -199,7 +199,44 @@ def GA_GraphColoringInts(edges, fitness):
     population = init_population(8, [0, 1], 4)
 
     return genetic_algorithm(population, fitness)
-
+def test_simpleProblemSolvingAgent():
+    class vacuumAgent(SimpleProblemSolvingAgentProgram):
+        def update_state(self, state, percept):
```

Comment:
I am not sure why you take `state` as a parameter. The parent class defines the function as `update_state(self, percept)`. Can't you just return `percept`?

Reply:
The parent class does defines it like this but in the __call__ method the function is passed with both of the arguments. I don't think there is a need for that. Should I keep it like this or change the parent class too? @MrDupin

Reply:
Ah yes, that is true. I would keep `def update_state(self, state, percept)`, but I would just return `percept` to keep things simple.